src/deserialize.rs

```diff
@@ -0,0 +1,173 @@
+use crate::array::make_mutable_arrays;
+use crate::reader::{get_schema, read_from_capnp_struct};
+use crate::zipped_field::{zip_fields, ZippedField};
+use capnp::Error;
+use capnp::{dynamic_struct, dynamic_value};
+use polars_arrow::array::{
+    Array, MutableArray, MutableBinaryArray, MutableBooleanArray, MutableDictionaryArray,
+    MutableListArray, MutablePrimitiveArray, MutableStructArray, MutableUtf8Array, TryPush,
+};
+use polars_arrow::chunk::Chunk;
+use polars_arrow::datatypes::{ArrowDataType, Field as ArrowField};
+
+macro_rules! push_value {
+    ($a:expr, $t:ty, $v:expr) => {{
+        let array = $a.as_mut_any().downcast_mut::<$t>().unwrap();
+        array.push(Some($v));
+    }};
+}
+
+pub fn deserialize(
+    messages: &[dynamic_value::Reader],
+    arrow_fields: &[ArrowField],
+) -> Result<Chunk<Box<dyn Array>>, Error> {
+    let mut arrays = make_mutable_arrays(arrow_fields, messages.len());
+    let zipped_fields = zip_fields(get_schema(messages), arrow_fields).unwrap();
```

Comment:
Would it be reasonable to handle possible errors here rather than using `unwrap`?

Reply:
I think it makes sense to convert capnp errors into PolarsError [similar to avro errors](https://github.com/pola-rs/polars/blob/cf4df740cabb5662866d1e9f450b5368aafda2ae/crates/polars-error/src/lib.rs#L94-L99). Then we can start using `PolarsResult` and `?` everywhere. I am hoping to try to contribute this directly to the polars project and make that change there.